src/options.rs

```diff
@@ -318,53 +321,80 @@ fn get_matches(app: clap::App) -> clap::ArgMatches {
         Err(e) => {
             match e.kind {
                 UnknownArgument | InvalidSubcommand => {
-                    // Make sure we're only dealing with the first-level command
-                    if let Some(first_cmd) = env::args_os().skip(1).next() {
-                        if first_cmd != e.info[0][..] {
-                            e.exit();
-                        }
+                    let mismatch_cmd = &e.info[0][..];
+                    match get_deprecated_matches(mismatch_cmd) {
+                        Some(matches) => matches,
+                        None => e.exit(),
                     }
-                    let new_name = match &e.info[0][..] {
-                         "create-database" => "database create",
-                         "create-migration" => "migration create",
-                         "list-aliases" => "list aliases",
-                         "list-casts" => "list casts",
-                         "list-databases" => "list databases",
-                         "list-indexes" => "list indexes",
-                         "list-modules" => "list modules",
-                         "list-object-types" => "list types",
-                         "list-scalar-types" => "list scalars",
-                         "list-roles" => "list roles",
-                         "migration-log" => "migration log",
-                         "self-upgrade" => "self upgrade",
-                         "show-status" => "migration status",
-                         _ => e.exit(),
-                    };
-                    let error = "warning:".bold().light_yellow();
-                    let cmd = e.info[0][..].green();
-                    let instead = format!("edgedb {}", new_name).green();
-                    eprintln!("\
-                        {error} The subcommand '{cmd}' was renamed\n\
-                        \n         \
-                            Use '{instead}' instead\
-                        \n\
-                    ", error=error, cmd=cmd, instead=instead);
-                    let new_args: Vec<OsString> = env::args_os().take(1).chain(
-                        new_name.split(" ").map(|x|x.into())
-                    ).chain(
-                        env::args_os().skip(2)
-                    ).collect();
-                    let app = <RawOptions as clap::IntoApp>::into_app();
-                    let app = update_help(app);
-                    return app.get_matches_from(new_args);
                 }
-                _ => {}
+                _ => e.exit(),
             }
-            e.exit();
         }
     }
 }
 
+fn get_deprecated_matches(mismatch_cmd: &str) -> Option<clap::ArgMatches> {
+    let mut args = env::args_os().skip(1);
```

Comment:
Unfortunately this may be a problem in case `edgedb --something instance init` or even `edgedb instance --something init`. I'm not sure how practical these uses currently, though.

Reply:
Ah you're right - let me see how to fix this.